### ISSUE TITLE
Methods with overload are parsed / checked incorrectly

### ISSUE THREAD

**Issue by davudkurbanov** [Riverside-Software/sonar-openedge#1059]
Hello

Today I noticed a weird in our sonar web-interface about possible value overflow:
![image](https://github.com/Riverside-Software/sonar-openedge/assets/55686942/ee6052b9-0b54-462d-9b4a-ee88cc897261)

`HTMLValue` is of datatype `CHARACTER`

In the `HTMLStripper` class I have a method with 2 overloads:
 - `METHOD PUBLIC STATIC LONGCHAR HtmlEscape(InputHtml AS LONGCHAR)`
 - `METHOD PUBLIC STATIC CHARACTER HtmlEscape(InputHtml AS CHARACTER):`

<sub> It goes without saying that whenever a `CHARACTER` value is required, then the `CHARACTER` overload is used, otherwise the `LONGCHAR` overload. </sub>

I added some code to provide more logging which function is used at the runtime:
![image](https://github.com/Riverside-Software/sonar-openedge/assets/55686942/413e6356-2190-4cee-ab58-20aadbb248b3)

So at runtime this works as expected and there is no possibility of overflow here. What could be the issue?
If you need any additional info, then contact me about this issue.

**Comment by gquerret** [comment]
Confirmed and reproduced internally.

**Comment by dominickdm** [comment]
Similar issue? SetMetaCharsetUtf8Tag returns LONCHAR and is assigned to a CHARACTER, should also raise this issue, no?

![image](https://github.com/Riverside-Software/sonar-openedge/assets/40289256/3c8cdc08-85f6-484d-a378-9238cd0d4791)

**Comment by gquerret** [comment]
@dominickdm There's another issue that has been reported that is related to this rule in SonarLint. That's also in the backlog.

**Comment by gquerret** [comment]
Depends on #1068

**Comment by gquerret** [comment]
Fixed in develop
